src/traversability_boundary.cpp

```diff
@@ -0,0 +1,592 @@
+#include "spark_dsg/traversability_boundary.h"
+
+namespace spark_dsg {
+
+bool isTraversable(TraversabilityState state, bool optimistic) {
+  return state == TraversabilityState::TRAVERSABLE ||
+         state == TraversabilityState::TRAVERSED ||
+         (optimistic && state == TraversabilityState::UNKNOWN);
+}
+
+bool isTraversable(const TraversabilityStates& states, bool optimistic) {
+  for (const auto& state : states) {
+    if (!isTraversable(state, optimistic)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+bool isIntraversable(TraversabilityState state, bool optimistic) {
+  return state == TraversabilityState::INTRAVERSABLE ||
+         (optimistic && state == TraversabilityState::UNKNOWN);
```

Comment:
Shouldn't this clause be the inverse (i.e., `!optimistic && state == TraversabilityState::UNKNOWN`)?

Reply:
Possible I misunderstood the docstring though